modules/olm-configuring-aws-sts.adoc

```diff
@@ -0,0 +1,213 @@
++
+[source,terminal]
+----
+$ chmod 775 ccoctl
+----
+
+.. Optional: Delete the pull secret file by running the following command:
++
+[source,terminal]
+----
+$ rm ./pull-secret.json
+----
+
+ifndef::olm-generic[]
+. Extract the `CredentialsRequest` custom resource (CR) that the Operator generated during installation by running the following command:
++
+[source,terminal]
+----
+$ oc get credentialsrequest \
+  -n <operator_cr_namespace> <operator_cr_name> \
+  -o yaml > <path_to_directory_with_list_of_credentials_requests>/credrequests/<operator_cr_name>.yaml
+----
+where:
++
+--
+** `<operator_cr_namespace>` is the namespace in which the Operator generates the CR.
+** `<operator_cr_name>` is the name of the CR the Operator generates.
+--
+endif::[]
+
+ifdef::olm-generic[]
+. Create and save a `CredentialsRequest` custom resource (CR) YAML file, such as shown in the following example, and then place it in the `credrequests` directory:
++
+.Example
++
+[source, yaml]
+----
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: <operator_cr_name> <1>
+  namespace: <operator_cr_namespace> <2>
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries: <3>
+    - action:
+      - iam:ListAccessKeys:*
+      effect: Allow
+      resource: 'arn:aws:s3:::*'
+  secretRef: 
+    name: <operator_secret_name> <4>
+    namespace: <operator_secret_namespace> <5>
+  serviceAccountNames:
+  - <service_account_name> <6>
+----
+<1> Specify a name for the CR. For easier reference, choose a name related to the name of the Operator.
+<2> Specify a namespace for the CR. For easier reference, choose a name related to the name of the Operator.
+<3> Specify IAM JSON policy statement entries. The values in this example are provided for formatting clarification only and must be updated for the requirements of the Operator. For more information, see AWS documentation about link:https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_statement.html[IAM JSON policy elements: Statement].
+<4> Specify a name for the secret that will be used by the Operator. For easier reference, choose a name related to the name of the Operator.
+<5> Specify the namespace in which the Operator that consumes the AWS credentials is deployed. For easier reference, choose a name related to the name of the Operator.
+<6> Specify the name of the service account that is used by the Operator.
+endif::[]
+
+. Use the `ccoctl` tool to process all `CredentialsRequest` objects in the `credrequests` directory by running the following command:
++
+[source, terminal]
+----
+$ ccoctl aws create-iam-roles \
+  --name=<name> \
+  --region=<aws_region> \
+  --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>/credrequests \ 
+  --identity-provider-arn=arn:aws:iam::<aws_account_id>:oidc-provider/<cluster_name>-oidc.s3.<aws_region>.amazonaws.com
```

Comment:
I would add one step before this to retrieve the `issuerURL` from the Authentication object, instead of predicting this value.
This value can be changed/customized and a recently added new option using CloudFront will break this pattern.
wdyt @abutcher ?

Reply:
This makes sense to me. For example, the `serviceAccountIssuer` within the authentication object could be grepped like,

```
$ oc get authentication cluster -o yaml | grep serviceAccountIssuer
  serviceAccountIssuer: https://abutcher-test-oidc.s3.us-east-1.amazonaws.com
```

Or the value of `.spec.serviceAccountIssuer` could be pulled out directly like,

```
$ oc get authentication cluster -o jsonpath --template='{ .spec.serviceAccountIssuer }'     
https://abutcher-test-oidc.s3.us-east-1.amazonaws.com
```

Reply:
Thanks @abutcher .
@jeana-redhat ^ it should be a better approach to be used on the `--identity-provider-arn` option.